File: src/main/java/fi/finlit/edith/ui/components/AuthAwarePanel.java
/*
 * Copyright (c) 2009 Mysema Ltd.
 * All rights reserved.
 * 
 */
package fi.finlit.edith.ui.components;

import org.apache.tapestry5.ioc.annotations.Inject;

import fi.finlit.edith.ui.services.AuthService;

/**
 * AuthAwarePanel provides
 *
 * @author tiwe
 * @version $Id$
 */
public abstract class AuthAwarePanel {
    
    @Inject
    private AuthService authService;
    
    public boolean isLoggedIn(){
        return authService.isAuthenticated();        
    }

    public String getUsername(){
        return authService.getUsername();
    }

}

File: src/main/scala/fi/finlit/edith/ui/components/AuthAwarePanel.scala
package fi.finlit.edith.ui.components

import org.apache.tapestry5.ioc.annotations.Inject
import fi.finlit.edith.ui.services.AuthService
//remove if not needed
import scala.collection.JavaConversions._

abstract class AuthAwarePanel {

  @Inject
  private var authService: AuthService = _

  def isLoggedIn(): Boolean = authService.isAuthenticated

  def getUsername(): String = authService.getUsername
}
